e2e/interchaintestv8/chainconfig/kurtosis.go

```diff
@@ -0,0 +1,199 @@
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/strangelove-ventures/interchaintest/v8/testutil"
+
+	"github.com/srdtrk/solidity-ibc-eureka/e2e/v8/ethereum"
+)
+
+const FaucetPrivateKey = "0x4b9f63ecf84210c5366c66d68fa1f5da1fa4f634fad6dfc86178e4d79ff9e59"
+
+type EthKurtosisChain struct {
+	RPC             string
+	BeaconApiClient ethereum.BeaconAPIClient
+	Faucet          *ecdsa.PrivateKey
+
+	kurtosisCtx *kurtosis_context.KurtosisContext
+	enclaveCtx  *enclaves.EnclaveContext
+}
+
+type kurtosisNetworkParams struct {
+	Participants        []kurtosisParticipant       `json:"participants"`
+	NetworkParams       kurtosisNetworkConfigParams `json:"network_params"`
+	WaitForFinalization bool                        `json:"wait_for_finalization"`
+}
+
+type kurtosisParticipant struct {
+	CLType     string `json:"cl_type"`
+	ELImage    string `json:"el_image"`
+	ELLogLevel string `json:"el_log_level"`
+}
+
+type kurtosisNetworkConfigParams struct {
+	Preset string `json:"preset"`
+}
+
+func SpinUpKurtosisPoS(ctx context.Context) (EthKurtosisChain, error) {
+	faucet, err := crypto.ToECDSA(ethcommon.FromHex(FaucetPrivateKey))
+	if err != nil {
+		return EthKurtosisChain{}, err
+	}
+
+	kurtosisCtx, err := kurtosis_context.NewKurtosisContextFromLocalEngine()
+	if err != nil {
+		return EthKurtosisChain{}, err
+	}
+
+	enclaveName := "ethereum-pos-testnet"
+	enclaves, err := kurtosisCtx.GetEnclaves(ctx)
+	if err != nil {
+		return EthKurtosisChain{}, err
+	}
+
+	if enclaveInfos, found := enclaves.GetEnclavesByName()[enclaveName]; found {
+		for _, enclaveInfo := range enclaveInfos {
+			err = kurtosisCtx.DestroyEnclave(ctx, enclaveInfo.EnclaveUuid)
+			if err != nil {
+				return EthKurtosisChain{}, err
+			}
+		}
+	}
+	enclaveCtx, err := kurtosisCtx.CreateEnclave(ctx, enclaveName)
+	if err != nil {
+		return EthKurtosisChain{}, err
+	}
+	networkParams := kurtosisNetworkParams{
+		Participants: []kurtosisParticipant{
+			{
+				CLType:     "lodestar",
+				ELImage:    "ethereum/client-go:v1.14.6",
+				ELLogLevel: "info",
+			},
+		},
+		NetworkParams: kurtosisNetworkConfigParams{
+			Preset: "minimal",
+		},
+		WaitForFinalization: true,
+	}
+	networkParamsJson, err := json.Marshal(networkParams)
+	if err != nil {
+		return EthKurtosisChain{}, err
+	}
+	starlarkResp, err := enclaveCtx.RunStarlarkRemotePackageBlocking(ctx, "github.com/ethpandaops/ethereum-package", &starlark_run_config.StarlarkRunConfig{
+		SerializedParams: string(networkParamsJson),
+	})
+	if err != nil {
+		return EthKurtosisChain{}, err
+	}
+	fmt.Println(starlarkResp.RunOutput)
+
+	gethCtx, err := enclaveCtx.GetServiceContext("el-1-geth-lodestar")
+	if err != nil {
+		return EthKurtosisChain{}, err
+	}
+	rpcPortSpec := gethCtx.GetPublicPorts()["rpc"]
+	rpc := fmt.Sprintf("http://localhost:%d", rpcPortSpec.GetNumber())
+
+	lighthouseCtx, err := enclaveCtx.GetServiceContext("cl-1-lodestar-geth")
+	if err != nil {
+		return EthKurtosisChain{}, err
+	}
```

Comment:
It would be nice to add some explaining comments to these.

Reply:
Good point, will do that!